libdevcore/Whiskers.cpp

```diff
@@ -106,51 +106,66 @@ string Whiskers::replace(
 	map<string, vector<StringMap>> const& _listParameters
 )
 {
-	using namespace boost;
-	static regex listOrTag("<(" + paramRegex() + ")>|<#(" + paramRegex() + ")>(.*?)</\\2>|<\\?(" + paramRegex() + ")>(.*?)(<!\\4>(.*?))?</\\4>");
-	return regex_replace(_template, listOrTag, [&](match_results<string::const_iterator> _match) -> string
+	static regex listOrTag("<(" + paramRegex() + ")>|<#(" + paramRegex() + ")>((?:.|\\r|\\n)*?)</\\2>|<\\?(" + paramRegex() + ")>((?:.|\\r|\\n)*?)(<!\\4>((?:.|\\r|\\n)*?))?</\\4>");
+	sregex_iterator curMatch(_template.begin(), _template.end(), listOrTag);
+	sregex_iterator matchEnd;
+
+	string result;
+	result.reserve(_template.length());
+	string::const_iterator lastMatchedPos(_template.cbegin());
+	while (curMatch != matchEnd)
 	{
-		string tagName(_match[1]);
-		string listName(_match[2]);
-		string conditionName(_match[4]);
-		if (!tagName.empty())
+		result.append(curMatch->prefix().first, curMatch->prefix().second);
+		result.append(([&](match_results<string::const_iterator> _match) -> string
```

Comment:
i think it might be less confusing to add an implementation of `boost::regex_replace` using `std::regex`. What do you think?

Reply:
Yes, that would make logic easier to understand. I will try it shortly :)